core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -82,7 +82,7 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
    * `F[A]` values.
    */
   def recover[A](fa: F[A])(pf: PartialFunction[E, A]): F[A] =
-    handleErrorWith(fa)(e => (pf.andThen(pure)).applyOrElse(e, raiseError))
+    handleErrorWith(fa)(e => (pf.andThen(pure _)).applyOrElse(e, raiseError _))
```

Comment:
may i ask why those underscores are needed now? What have changed to make them required?

Reply:
I think it's https://github.com/scala/scala/pull/7263